source/Detector/Amptek/AmptekSDD123ServerGroup.cpp

```diff
@@ -0,0 +1,166 @@
+	:QObject(parent)
+{
+	amptekServerGroupThread_ = new QThread();
+
+	amptekServerGroup_ = new AmptekSDD123ServerGroup(configurationMaps);
+	amptekServerGroup_->moveToThread(amptekServerGroupThread_);
+
+	connect(amptekServerGroupThread_, SIGNAL(finished()), amptekServerGroup_, SLOT(deleteLater()));
+
+	connect(amptekServerGroup_, SIGNAL(serverChangedToConfigurationState(int)), this, SIGNAL(serverChangedToConfigurationState(int)));
+	connect(amptekServerGroup_, SIGNAL(serverChangedToDwellState(int)), this, SIGNAL(serverChangedToDwellState(int)));
+
+	amptekServerGroupThread_->start();
+}
+
+AmptekSDD123ThreadedDataServerGroup::~AmptekSDD123ThreadedDataServerGroup()
+{
+	if (amptekServerGroupThread_->isRunning()) {
+		amptekServerGroupThread_->quit();
+	}
+
+	amptekServerGroupThread_->deleteLater();
+}
+
+AmptekSDD123Server*  AmptekSDD123ThreadedDataServerGroup::serverAt(int index)
+{
+	return amptekServerGroup_->serverAt(index);
+}
+
+
+/// =================================== implementation of AmptekSDD123ServerGroup =======================================
+AmptekSDD123ServerGroup::AmptekSDD123ServerGroup(QList<AmptekSDD123ConfigurationMap *> configurationMaps, QObject *parent) :
+	QObject(parent)
+{
+	requestSpectraTimer_ = 0;
+	masterRequestInterval_ = 50;
+//	serverState_ = AmptekSDD123ServerGroup::DwellState;
+
+	serverAboutToConfigurationStateMapper_ = new QSignalMapper();
+	serverConfigurationStateMapper_ = new QSignalMapper();
+	serverDwellStateMapper_ = new QSignalMapper();
+
+	AmptekSDD123Server *amptekServer;
+	for(int index = 0, size = configurationMaps.count(); index < size; index++){
+		AmptekSDD123ConfigurationMap *amptekConfiguration = configurationMaps.at(index);
+
+		amptekServer = new AmptekSDD123Server(amptekConfiguration, QString("%1 Server").arg(amptekConfiguration->detectorName()), this);
+		amptekServer->setRequestIntervalTimer(masterRequestInterval_);
+
+		servers_.append(amptekServer);
+		serversInDwellMode_.append(amptekServer);
+
+		serverAboutToConfigurationStateMapper_->setMapping(amptekServer, index);
+		serverConfigurationStateMapper_->setMapping(amptekServer, index);
+		serverDwellStateMapper_->setMapping(amptekServer, index);
+
+		connect(amptekServer, SIGNAL(serverAboutToChangeToConfigurationState()), serverAboutToConfigurationStateMapper_, SLOT(map()));
+		connect(amptekServer, SIGNAL(serverChangedToConfigurationState()), serverConfigurationStateMapper_, SLOT(map()));
+		connect(amptekServer, SIGNAL(serverChangedToDwellState()), serverDwellStateMapper_, SLOT(map()));
+	}
+
+	connect(serverAboutToConfigurationStateMapper_, SIGNAL(mapped(int)), this, SLOT(onServerAboutToChangeToConfigurationState(int)));
+	connect(serverConfigurationStateMapper_, SIGNAL(mapped(int)), this, SLOT(onServerChangedToConfigurationState(int)));
+	connect(serverDwellStateMapper_, SIGNAL(mapped(int)), this, SLOT(onServerChangedToDwellState(int)));
+
+	QTimer::singleShot(1000, this, SLOT(initiateAllRequestSpectrum()));
```

Comment:
I'm assuming this is a legacy thing, but what is the specific reason for 1 second?